target/linux/mediatek/dts/mt7981b-yuncore-ax835.dts

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+
+/dts-v1/;
+
+#include "mt7981.dtsi"
+
+/ {
+	compatible = "yuncore,ax835", "mediatek,mt7981";
+	model = "YunCore AX835";
+
+	aliases {
+		ethernet0 = &gmac0;
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		serial0 = &uart0;
+	};
+
+	chosen {
+		// TODO from stock:
+		// bootargs = ".. loglevel=8 earlycon=uart8250,mmio32,0x11002000";
+		// u-boot,bootconf = "config-1";
+		stdout-path = "serial0:115200n8";
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&pio 1 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		// TODO
+		// GPIO 5 is required to be high to allow other LEDs to function.
+		// atm, this happens to be the case but an explicit setting would be nicer.
+		// in stock it's arranged like this:
+		// onekey: gpio5 active_high (JTAG JTDI - according to spreadsheet)
+		// "Higher level circuit to enable all LED light; Lower level circuit to close"
+
+		led_system: led@0 {
+			label = "red:system";
+			gpios = <&pio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		led@1 {
+			label = "green:wifi2";
+			gpios = <&pio 34 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led@2 {
+			label = "blue:wifi5";
+			gpios = <&pio 35 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		led@3 {
+			// a gpio-wdt watchdog couldn't be made to work.
+			// the device rebooted after 5 minutes.
+			label = "hwwatchdog";
+			gpios = <&pio 7 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "timer";
+			led-pattern = <1000>, <1000>;
+		};
+
+		// TODO
+		// there's another watchdog gpio in the spreadsheet:
+		// syswatchdog: gpio2 (covered by that from mt7981.dtsi?)
```

Comment:
If there is an external watchdog, you need a watchdog driver for this. Maybe this works: https://elixir.bootlin.com/linux/v6.1.63/source/drivers/watchdog/gpio_wdt.c

Reply:
I tried that and it doesn't work - for some reason I don't understand:

```
	watchdog {
		compatible = "linux,wdt-gpio";
		gpios = <&pio 7 GPIO_ACTIVE_LOW>;
		hw_algo = "toggle";
		hw_margin_ms = <1000>;
		always-running;
	};
```

It just still reboots. I'd love to know why, though :thinking: